effects/game-of-life/game-of-life.c

```diff
@@ -0,0 +1,445 @@
+//
+static void BlitAdjacentHorizontal(const BitmapT* source, BitmapT* target, u_short minterms)
+{
+  void* srcCenter = source->planes[0] + source->bytesPerRow; // C channel
+  void* srcRight = source->planes[0] + source->bytesPerRow + 2; // B channel
+  void* srcLeft = source->planes[0] + source->bytesPerRow; // A channel
+  u_short bltsize = (BOARD_HEIGHT << 6) | ((EXT_BOARD_WIDTH / 16));
+
+  custom->bltcon0 = ASHIFT(1) | minterms | (SRCA | SRCB | SRCC | DEST);
+  custom->bltcon1 = BSHIFT(15);
+  
+  custom->bltafwm = 0x0000;
+  custom->bltalwm = 0x0000;
+  custom->bltamod = 0;
+  custom->bltbmod = 0;
+  custom->bltcmod = 0;
+  custom->bltdmod = 0;
+  custom->bltapt = srcLeft;
+  custom->bltbpt = srcRight;
+  custom->bltcpt = srcCenter;
+  custom->bltdpt = target->planes[0] + target->bytesPerRow;
+  custom->bltsize = bltsize;
+}
+
+// setup blitter to calculate a function of three vertically adjacent lit pixels
+static void BlitAdjacentVertical(const BitmapT* source, BitmapT* target, u_short minterms)
+{
+  void* srcCenter = source->planes[0] + source->bytesPerRow; // C channel
+  void* srcUp = source->planes[0]; // A channel
+  void* srcDown = source->planes[0] + 2*source->bytesPerRow; // B channel
+  u_short bltsize = (BOARD_HEIGHT << 6) | ((EXT_BOARD_WIDTH / 16));
+
+  custom->bltcon0 = minterms | (SRCA | SRCB | SRCC | DEST);
+  custom->bltcon1 = 0;
+  
+  custom->bltafwm = 0xFFFF;
+  custom->bltalwm = 0xFFFF;
+  custom->bltamod = 0;
+  custom->bltbmod = 0;
+  custom->bltcmod = 0;
+  custom->bltdmod = 0;
+  custom->bltapt = srcUp;
+  custom->bltbpt = srcDown;
+  custom->bltcpt = srcCenter;
+  custom->bltdpt = target->planes[0] + target->bytesPerRow;
+  custom->bltsize = bltsize;
+}
+
+// setup blitter for a standard blit without shifts
+static void BlitFunc(const BitmapT* sourceA, const BitmapT* sourceB, const BitmapT* sourceC, const BitmapT* target, u_short minterms)
+{
+  u_short bltsize = (BOARD_HEIGHT << 6) | ((EXT_BOARD_WIDTH / 16));
+
+  custom->bltcon0 = minterms | (SRCA | SRCB | SRCC | DEST);
+  custom->bltcon1 = 0;
+  
+  custom->bltafwm = 0xFFFF;
+  custom->bltalwm = 0xFFFF;
+  custom->bltamod = 0;
+  custom->bltbmod = 0;
+  custom->bltcmod = 0;
+  custom->bltdmod = 0;
+  custom->bltapt = sourceA->planes[0] + sourceA->bytesPerRow;
+  custom->bltbpt = sourceB->planes[0] + sourceB->bytesPerRow;
+  custom->bltcpt = sourceC->planes[0] + sourceC->bytesPerRow;
+  custom->bltdpt = target->planes[0] + target->bytesPerRow;
+  custom->bltsize = bltsize;
+}
+
+static void (*PixelDouble)(u_char* source asm("a0"), u_short* target asm("a1"), u_short* lut asm("a2"));
+#define PixelDoubleSize (BOARD_WIDTH * BOARD_HEIGHT * 10 + BOARD_HEIGHT * 2 + 6)
+
+// doubles pixels horizontally
+static void MakePixelDoublingCode(const BitmapT* bitmap)
+{
+  u_short x;
+  u_short y;
+  u_short *code = (void*)PixelDouble;
+
+  *code++ = 0x7200; // moveq #0,d1
+  *code++ = 0x7400 | (EXT_BOARD_MODULO & 0xFF); // moveq #EXT_BOARD_MODULO,d2
```

Comment:
Since `d2` is callee saved register you should not modify it without saving it on stack in your speed code.